optimum/exporters/tasks.py

```diff
@@ -1161,11 +1164,26 @@ def determine_framework(
                 for file in filenames
             ]
         else:
-            if not isinstance(model_name_or_path, str):
-                model_name_or_path = str(model_name_or_path)
-            all_files = huggingface_hub.list_repo_files(model_name_or_path, repo_type="model")
-            if subfolder != "":
-                all_files = [file[len(subfolder) + 1 :] for file in all_files if file.startswith(subfolder)]
+            object_id = model_name_or_path.replace("/", "--")
+            full_model_path = Path(cache_dir, f"models--{object_id}")
+            if full_model_path.is_dir():  # explore the cache first
+                # Resolve refs (for instance to convert main to the associated commit sha)
+                revision_file = Path(full_model_path, "refs", "main")
+                if revision_file.is_file():
+                    with open(revision_file) as f:
+                        revision = f.read()
+                cached_path = Path(full_model_path, "snapshots", revision, subfolder)
+                all_files = [
+                    os.path.relpath(os.path.join(dirpath, file), cached_path)
```

Comment:
nit: We should stick to the pathlib API here instead of mixing pathlib and os.path IMO

Reply:
agreed